File: app/core/src/main/java/com/fsck/k9/message/extractors/AttachmentCounter.java
package com.fsck.k9.message.extractors;


import java.util.ArrayList;
import java.util.List;

import com.fsck.k9.mail.Message;
import com.fsck.k9.mail.MessagingException;
import com.fsck.k9.mail.Part;
import com.fsck.k9.mail.internet.MessageExtractor;


public class AttachmentCounter {

    public static AttachmentCounter newInstance() {
        return new AttachmentCounter();
    }

    public int getAttachmentCount(Message message) throws MessagingException {
        List<Part> attachmentParts = new ArrayList<>();
        MessageExtractor.findViewablesAndAttachments(message, null, attachmentParts);

        return attachmentParts.size();
    }
}

File: legacy/core/src/main/java/com/fsck/k9/message/extractors/AttachmentCounter.kt
package com.fsck.k9.message.extractors

import com.fsck.k9.mail.Message
import com.fsck.k9.mail.Part
import com.fsck.k9.mail.internet.MessageExtractor
import net.thunderbird.core.common.exception.MessagingException

class AttachmentCounter {
    @Throws(MessagingException::class)
    fun getAttachmentCount(message: Message): Int {
        val attachmentParts = ArrayList<Part>()
        MessageExtractor.findViewablesAndAttachments(
            message,
            null,
            attachmentParts,
        )

        return attachmentParts.size
    }

    companion object {
        fun newInstance() = AttachmentCounter()
    }
}
